src/dmd/semantic2.d

```diff
@@ -558,14 +569,18 @@ private extern(C++) final class Semantic2Visitor : Visitor
                         if (e.op == TOK.tuple)
                         {
                             TupleExp te = cast(TupleExp)e;
-                            eval(sc, te.exps);
+                            eval(sc, te.exps, lastTag);
                         }
+
+                        // Handles compiler-recognized `core.attribute.gnuAbiTag`
+                        if (UserAttributeDeclaration.isGNUABITag(e))
+                            doGNUABITagSemantic(e, lastTag);
```

Comment:
The condition could be moved to either Compiler or Target (probably the former is best) so other compilers can extend their own compiler recognised udas - if I understand that this is the correct place to do so.

Reply:
Ah wait, I didn't get to the bit where it's a library defined uda.

In which case, there should be tests that ensure bad things don't happen if people supply their own (incorrect) type.